### PR TITLE
fix(DB/Creature): Add correct stealth to Crag Stalker

### PR DESCRIPTION
**Pull request by Si1ker**
## Issues Addressed:

- Closes https://github.com/chromiecraft/chromiecraft/issues/5662
Progress at https://github.com/azerothcore/azerothcore-wotlk/issues/16549

## SOURCE:

https://www.wowhead.com/wotlk/npc=4126/crag-stalker#abilities
https://github.com/TrinityCore/TrinityCore/commit/4e8b0fcaa584f764089a3cec1bd7165c3d0c0659

Co-authored by: CraftedRO <CraftedRO@users.noreply.github.com>

## Tests Performed:

image

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by heyitsbench** [comment]
> * Partially closes 16549

You may want to re-word this to something like 'Mkaes progress towards 16549'. As it currently is, merging this PR will automatically close the issue.

**Comment by Si1ker** [comment]
> > * Partially closes 16549
> 
> You may want to re-word this to something like 'Makes progress towards 16549'. As it currently is, merging this PR will automatically close the issue.

Ok, that's correct.

side serious question Bench, do you have a tracker in my activity? lol

**Comment by heyitsbench** [comment]
> side serious question Bench, do you have a tracker in my activity? lol

I just pay attention. ❤️

**Comment by Kitzunu** [comment]
Bench as a microscope on us all ❤️

**Comment by Gultask** [comment]
Maybe for surprise attack, add a condition on the object so it's only cast if it has the stealth aura?

**Comment by Si1ker** [comment]
so, can a one-line sai switching on spell ID for another and its even_type be reviewed? o.o

**Comment by Kitzunu** [comment]
Add correct co-author to your commit
`CraftedRO <CraftedRO@users.noreply.github.com>`

File: data/sql/updates/pending_db_world/rev_1687028319280087800.sql [hunk 1]
```diff
@@ -1,5 +1,6 @@
 --
-DELETE FROM `smart_scripts` WHERE `entryorguid` = 4126 AND `source_type` = 0 AND `id` = 0;
-DELETE FROM `creature_template_addon` WHERE `entry` = 4126;
-INSERT INTO `creature_template_addon` (`entry`, `path_id`, `mount`, `bytes1`, `bytes2`, `emote`, `visibilityDistanceType`, `auras`) VALUES (4126, 0, 0, 0, 1, 0, 0, '22766');
+UPDATE `smart_scripts` SET `event_flags`=0, `action_param1`=22766, `comment`='Crag Stalker - Out of Combat - Cast \'Stealth\'' WHERE  `entryorguid`=4126 AND `source_type`=0 AND `id`=0 AND `link`=0;
```
Comment (Gultask) [inline data/sql/updates/pending_db_world/rev_1687028319280087800.sql:2]: Use ON_RESET or template_addon instead

File: data/sql/updates/pending_db_world/rev_1687028319280087800.sql [hunk 2]
```diff
@@ -1,5 +1,6 @@
...
+UPDATE `smart_scripts` SET `event_type`=4, `event_flags`=0, `event_param1`=0, `event_param2`=0, `comment`='Crag Stalker - On Aggro - Cast \'Surprise Attack\'' WHERE  `entryorguid`=4126 AND `source_type`=0 AND `id`=1 AND `link`=0;
+DELETE FROM `smart_scripts` WHERE `entryorguid`=4126 AND `source_type`=0 AND `id`=2 AND `link`=0;
+INSERT INTO `smart_scripts` (`entryorguid`, `id`, `event_type`, `event_param2`, `event_param3`, `event_param4`, `action_type`, `action_param1`, `target_type`, `comment`) VALUES (4126, 2, 9, 8, 2000, 4000, 11, 8151, 7, 'Crag Stalker - Within 0-8 Range - Cast \'Surprise Attack\'');
```
Comment (Gultask) [inline data/sql/updates/pending_db_world/rev_1687028319280087800.sql:5]: hmm, wasn't Surprise Attack supposed to be cast only when stealthed? 🤔

**Comment by GastonCK** [comment]
tag me whenever is ready to be tested

**Comment by Si1ker** [comment]
> tag me whenever is ready to be tested

ready to be tested/reviewed.

**Comment by GastonCK** [comment]
Stealthed but no surprise attack.
WoWScrnShot_062923_220400
WoWScrnShot_062923_220750

Also there is a bug in which sometimes when you aggro the crag stalker but you don't move, you just wait standing still, the beast will attack you without breaking stealth, you have to move in order to force the beast out of stealth.
WoWScrnShot_062923_221100

**Comment by Si1ker** [comment]
@GastonCK this pr is only to use the correct spell ID of stealth, that's it.

**Review by Gultask — APPROVED** [review]

**Comment by Gultask** [comment]
btw I can confirm finally that adding

```
Co-authored-by: CraftedRO <CraftedRO@users.noreply.github.com>
```
worked. Why Github makes this so hard idk 😵

**Pull request merged**